File: LeetCode Solutions/Arrays 101/Sort Array By Parity/Sort Array By Parity.cpp
class Solution {
public:
    vector<int> sortArrayByParity(vector<int>& A) {
        int left = 0, right = A.size() - 1, temp;
        while (left < right){
            if (A[left] % 2 == 1){
                temp = A[left];
                A[left] = A[right];
                A[right] = temp;
                right -= 1;
            } else{
                left += 1;
            }
        }
        return A;
    }
};

File: LeetCode Solutions/Arrays 101/Sort Array By Parity/Sort Array By Parity.py
class Solution:
    def sortArrayByParity(self, A: List[int]) -> List[int]:
        left, right = 0, len(A)-1
        while (left < right):
            if (A[left] % 2 == 1):
                A[left], A[right] = A[right], A[left]
                right -= 1
            else:
                left += 1
        return A